### PR TITLE
[FLINK-17827] [scala-shell] scala-shell.sh should fail early if no mo…

### PR DESCRIPTION
**Pull request by paul8263**
…de is specified, or have default logging settings

## What is the purpose of the change

Try to avoid the exception that log4j.properties and logback.xml are set wrongly when illegal mode or no mode is specified when running Flink Scala shell using start-scala-shell.sh

## Brief change log

- Added an additional if-else branch to set the default log4j.properties and logback.xml when  illegal mode or no mode is provided.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-scala-shell/start-script/start-scala-shell.sh [hunk 1]
```diff
@@ -86,11 +86,15 @@ if [[ ($MODE = "local") || ($MODE = "remote") ]]
 then
     LOG4J_CONFIG=log4j.properties
     LOGBACK_CONFIG=logback.xml
-elif [[ $1 = "yarn" ]]
+elif [[ $MODE = "yarn" ]]
```
Comment (zentol) [inline flink-scala-shell/start-script/start-scala-shell.sh:89]: maybe we could simplify this further:
```
if yarn {
  use the session files
else {
  use default files
}
```

**Review by zentol — CHANGES_REQUESTED** [review]
I think we can simplify this a bit further

**Comment by paul8263** [comment]
@zentol Thank you for your suggestion and now I changed my code. I would be glad if you take some time to review the updated solution.

File: flink-scala-shell/start-script/start-scala-shell.sh [hunk 2]
```diff
@@ -82,15 +82,15 @@ fi
...
 
-if [[ ($MODE = "local") || ($MODE = "remote") ]]
+if [[ $MODE = "yarn" ]]
 then
-    LOG4J_CONFIG=log4j.properties
-    LOGBACK_CONFIG=logback.xml
-elif [[ $1 = "yarn" ]]
-then
-    LOG4J_CONFIG=log4j-session.properties
+	LOG4J_CONFIG=log4j-session.properties
```
Comment (zentol) [inline flink-scala-shell/start-script/start-scala-shell.sh:87]:
```suggestion
    LOG4J_CONFIG=log4j-session.properties
```

File: flink-scala-shell/start-script/start-scala-shell.sh [hunk 3]
```diff
@@ -82,15 +82,15 @@ fi
...
     LOGBACK_CONFIG=logback-session.xml
     FLINK_CLASSPATH=$FLINK_CLASSPATH:$HADOOP_CLASSPATH:$HADOOP_CONF_DIR:$YARN_CONF_DIR
+else
+	# Set the default log config when MODE is something other than yarn. eg: local, remote or other invalid mode.
+	LOG4J_CONFIG=log4j.properties
```
Comment (zentol) [inline flink-scala-shell/start-script/start-scala-shell.sh:92]:
```suggestion
    LOG4J_CONFIG=log4j.properties
```

File: flink-scala-shell/start-script/start-scala-shell.sh [hunk 4]
```diff
@@ -82,15 +82,15 @@ fi
...
-    LOG4J_CONFIG=log4j.properties
-    LOGBACK_CONFIG=logback.xml
-elif [[ $1 = "yarn" ]]
+if [[ $MODE = "yarn" ]]
 then
     LOG4J_CONFIG=log4j-session.properties
     LOGBACK_CONFIG=logback-session.xml
     FLINK_CLASSPATH=$FLINK_CLASSPATH:$HADOOP_CLASSPATH:$HADOOP_CONF_DIR:$YARN_CONF_DIR
+else
+	# Set the default log config when MODE is something other than yarn. eg: local, remote or other invalid mode.
```
Comment (zentol) [inline flink-scala-shell/start-script/start-scala-shell.sh:91]:
```suggestion
    # Set the default log config when MODE is something other than yarn. eg: local, remote or other invalid mode.
```

**Pull request merged**